### PR TITLE
Add an option to number all displayed math

### PR DESCRIPTION
**Pull request by xuhdev**
The second commit also refactors the math extension tests to include tests for this new option (more elegantly).

### CODE REVIEW

**Comment by tk0miya** [comment]
LGTM. I'll merge this if no objections within a week.

File: sphinx/ext/mathbase.py [hunk 1]
```diff
@@ -38,7 +38,9 @@ def wrap_displaymath(math, label):
             ret.append('\\begin{split}%s\\end{split}' % part +
                        (label and '\\label{'+label+'}' or ''))
         else:
-            ret.append('\\begin{split}%s\\end{split}\\notag' % part)
+            ret.append('\\begin{split}%s\\end{split}' % part)
```
Comment (lehmannro) [inline sphinx/ext/mathbase.py]: Why not make both strings raw strings?

File: sphinx/ext/mathbase.py [hunk 2]
```diff
@@ -38,7 +38,9 @@ def wrap_displaymath(math, label):
             ret.append('\\begin{split}%s\\end{split}' % part +
                        (label and '\\label{'+label+'}' or ''))
         else:
-            ret.append('\\begin{split}%s\\end{split}\\notag' % part)
+            ret.append(r'\begin{split}%s\end{split}' % part)
```
Comment (xuhdev) [inline sphinx/ext/mathbase.py:41]: @lehmannro I've made the change. Thanks!

**Comment by tk0miya** [comment]
Oh, this PR has conflicted...
@xuhdev Could you rebase this? After that, I'll merge.

**Comment by xuhdev** [comment]
I've rebased the PR. Strange though I didn't get any conflict.

**Comment by tk0miya** [comment]
Merged. Thank you for contribution!

**Pull request merged**